### PR TITLE
Update apple silicon support

### PR DESCRIPTION
**Pull request by alexazl**
# Description
Update "Apple silicon support" section now that we have M1 support. Also add M1 Xcode support table to "Using macOS" page.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/using-macos.md [hunk 1]
```diff
@@ -66,9 +66,9 @@ To read about CircleCI's customer support policy regarding beta images, please c
 ### Apple silicon support
 {: #apple-silicon-support }
 
-It is possible to build Apple Silicon/universal binaries using Xcode `12.0.0` and higher, as Apple provides both the Intel (`x86_64`) and Apple Silicon (`arm64`) toolchains in this release. Cross-compiling Apple Silicon binaries on Intel hosts has an additional overhead, and, as a result, compilation times will be longer than native compilation for Intel.
+Running or testing Apple Silicon apps natively can be done on our silicon-based Macs or by using [CircleCI runner]({{site.baseurl}}/runner-overview/#supported).
```
Comment (kbatuigas) [inline jekyll/_cci2/using-macos.md:69]: > https://circleci.com/docs/runner-overview/#supported

We did a bit of a refresh on the runner content some time back and this path is no longer valid. Would you mind double checking the runner docs again and adding the URL to the section you want to include? Unfortunately it's not immediately evident to me just from reading the page again :\

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Just a clarifying comment

File: jekyll/_cci2/using-macos.md (hunk 1 above)
Comment (alexazl) [reply jekyll/_cci2/using-macos.md]: Updated! Thanks for pointing this out.

**Comment by alexazl** [comment]
@kbatuigas I've updated the Runner link. If everything looks good to you, please merge. I'll be out tomorrow and unable to merge it myself. If you have any other questions or comments, though, I'll take a look on Monday.

**Review by kbatuigas — COMMENTED** [review]
🎉

**Review by kbatuigas — APPROVED** [review]

**Pull request merged**